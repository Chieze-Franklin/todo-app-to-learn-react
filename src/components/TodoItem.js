import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { deleteTodo, toggleTodo } from '../actions/todoActions';

class TodoItemStyle {
  backgroundColor = '#f4f4f4'
  borderBottom = '1px #ccc dotted'
  padding = '10px'
}

class TodoItemCompletedStyle extends TodoItemStyle {
  textDecoration = 'line-through'
}

class TodoItemNotCompletedStyle extends TodoItemStyle {
  textDecoration = 'none'
}

const completedStyle = new TodoItemCompletedStyle();
const notCompletedStyle = new TodoItemNotCompletedStyle();
const btnStyle = {
  backgroundColor: '#ff0000',
  color: '#ffffff',
  boarder: 'none',
  padding: '5px 9px',
  borderRadius: '50%',
  cursor: 'pointer',
  float: 'right'
};

export class TodoItem extends Component {
  getStyle = () => this.props.todo.completed ? completedStyle : notCompletedStyle;
  render() {
    const { todo, deleteTodo, toggleTodo } = this.props;
    let modifiedTodo = this.props.todos.find(t => t.id === todo.id);
    const { id, completed, title } = modifiedTodo || todo;
    return (
      <div style={this.getStyle()}>
        <p>
          <input type="checkbox" onChange={toggleTodo.bind(this, id)} checked={completed} /> {' '}
          {title}
          <button style={btnStyle} onClick={deleteTodo.bind(this, id)}>x</button>
        </p>
      </div>
    )
  }
}

TodoItem.propTypes = {
  todo: PropTypes.object.isRequired,
  deleteTodo: PropTypes.func.isRequired,
  toggleTodo: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  deleteTodo,
  toggleTodo
};

const mapStateToProps = state => ({
  todos: state.todos.items
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoItem);
