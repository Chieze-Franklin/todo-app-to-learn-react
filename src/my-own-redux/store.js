import { createStore } from './index'

const reducer = (state, action) => {
  switch (action.type) {
    case 'PAGE_CHANGED':
      return {
        ...state,
        currentPage: action.payload
      }
    default:
      return state;
  }
}

const initialState = {
  appName: 'TodoList',
  currentPage: 'home',
}

export const changePage = (page) => (dispatch) => {
  dispatch({
    type: 'PAGE_CHANGED',
    payload: page
  })
};

export default createStore(reducer, initialState);
