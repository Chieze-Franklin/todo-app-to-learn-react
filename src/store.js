import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const initialState = {};

const middleware = [thunk];

const devtools = process.env.NODE_ENV === 'test'
  ? x => x /* eslint-disable no-underscore-dangle */
  : window.__REDUX_DEVTOOLS_EXTENSION__
      && window.__REDUX_DEVTOOLS_EXTENSION__();
/* eslint-enable no-underscore-dangle */

const store = createStore(rootReducer, initialState, compose(
  applyMiddleware(...middleware),
  devtools
));
// const store = compose(
//   applyMiddleware(...middleware),
//   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
// )(createStore)(rootReducer);
console.log(store)
export default store;