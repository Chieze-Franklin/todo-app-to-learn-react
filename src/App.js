import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Provider } from 'react-redux';

import './App.css';
import About from './components/pages/About';
import AddTodo from './components/AddTodoFunctionComponent';
import Header from './components/layout/Header';
import store from './store';
import Todos from './components/Todos';

import { Provider as AppContextProvider, Consumer } from './context';

class App extends Component {
  state = {
    appName: 'TodoList'
  }

  render() {
    return (
      <Provider store={store}>
        <Router>
          <div className="App">
            <div className="container">
              <AppContextProvider>
                <Header />
                <Route path="/" exact render={props => {
                  return (<Consumer>
                    {value => {
                      const { currentPage, dispatch } = value;
                      if (currentPage !== 'home') {
                        dispatch({
                          type: 'PAGE_CHANGED',
                          payload: 'home'
                        })
                      }
                      return (<React.Fragment>
                        <AddTodo />
                        <Todos />
                      </React.Fragment>)
                    }}
                  </Consumer>)}} />
                <Route path="/about" component={About} />
              </AppContextProvider>
            </div>
          </div>
        </Router>
      </Provider>
    );
  }
}

export default App;
