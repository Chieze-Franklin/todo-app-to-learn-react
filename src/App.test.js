import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from './store';

import App from './App';
import TodoItem from './components/TodoItem';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('TodoItem renders correctly', () => {
  const todo = {
    id: '1',
    title: 'Todo 1',
    completed: false
  }
  const tree = renderer
    .create(<Provider store={store}><TodoItem todo={todo} /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
