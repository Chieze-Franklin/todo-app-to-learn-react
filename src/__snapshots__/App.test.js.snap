// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders correctly 1`] = `
<div
  style={
    TodoItemNotCompletedStyle {
      "backgroundColor": "#f4f4f4",
      "borderBottom": "1px #ccc dotted",
      "padding": "10px",
      "textDecoration": "none",
    }
  }
>
  <p>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
     
     
    Todo 1
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#ff0000",
          "boarder": "none",
          "borderRadius": "50%",
          "color": "#ffffff",
          "cursor": "pointer",
          "float": "right",
          "padding": "5px 9px",
        }
      }
    >
      x
    </button>
  </p>
</div>
`;
